shell/platform/windows/text_input_plugin_unittest.cc

```diff
@@ -148,6 +206,109 @@ TEST(TextInputPluginTest, VerifyComposingSendStateUpdate) {
   EXPECT_TRUE(sent_message);
 }
 
+TEST(TextInputPluginTest, VerifyInputActionNewlineInsertNewLine) {
+  std::vector<std::vector<uint8_t>> messages;
+
+  TestBinaryMessenger messenger(
+      [&messages](const std::string& channel, const uint8_t* message,
+                  size_t message_size, BinaryReply reply) {
+        int length = static_cast<int>(message_size);
+        std::vector<uint8_t> last_message(length);
+        memcpy(&last_message[0], &message[0], length * sizeof(uint8_t));
+        messages.push_back(last_message);
```

Comment:
What do you think of storing `std::string`s instead?

```suggestion
        std::string last_message(reinterpret_cast<const char*>(message), message_size);
        messages.push_back(last_message);
```

Reply:
Would this still allow use of `std::equal` on https://github.com/flutter/engine/pull/42244/files/16ea94095b930f2c0a85e244b43abf31dbe690f0#diff-f913efa93fd3ac5c33b9fc924f41ff6e1a7f8300a5e548aa52fb0d30a3904b4cR247, as `update_state_message` will still be the same type?

Reply:
> What do you think of storing std::strings instead?

I choosed to rely on `std::vector<uint8_t>` because this is the type that is returned by `codec.EncodeMethodCall`, I think it is more understandable to stick to the type expectations of the existing APIs used in the test.

Reply:
Storing `std::string`s would make debugging a bit easier if things break.

If you still prefer storing `uint8_t`s, could you replace the `memcpy`'s `&last_message[0]` argument with  `last_message.data()`?

Reply:
Personally for line 248 I would just have the expected JSON stored as a string. Here's an example where we do that:

https://github.com/flutter/engine/blob/7c0a639efd0a0f6c9e260da739a80089be75e17b/shell/platform/fuchsia/flutter/text_delegate_unittests.cc#L228-L231